src/table/sticky-scrollbar/sticky-scrollbar.tsx

```diff
@@ -26,6 +27,12 @@ function StickyScrollbar(
   const scrollbarContentRef = React.useRef<HTMLDivElement>(null);
   const mergedRef = useMergeRefs(ref, scrollbarRef);
 
+  /**
+   * If the height of the scrollbar is 0, we're likely on a platform that uses
+   * overlay scrollbars (e.g. Mac).
+   */
+  const forceOffsetScrollbar = browserScrollbarSize().height === 0;
```

Comment:
Maybe it's worth just briefly checking with UI folks to make sure we're OK with applying this treatment to all tables.